var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -325,12 +325,12 @@ def setup_build_environment(self, env):
         if "+netcdf" in spec:
             # ESMF provides the ability to read Grid and Mesh data in
             # NetCDF format.
-            os.environ["ESMF_NETCDF"] = "nc-config"
-            os.environ["ESMF_NFCONFIG"] = "nf-config"
+            env.set["ESMF_NETCDF"] = "nc-config"
```

Comment:
So, the makefile knows about `nc-config` but you still have to call it below?

Reply:
Correct. ESMF's build system filters out much of the necessary output of `nc-config --libs`, so it doesn't handle static dependencies. I've discussed this with the developers and I believe it will be fixed in the future, but for now we either need to call `nc-config` ourselves or patch ESMF (and I think this is cleaner).